tapir/coop/tests/membership_resignation/test_form.py

```diff
@@ -21,30 +26,102 @@ def setUp(self) -> None:
+            "cancellation_date": resignation.cancellation_date,
+            "resignation_type": resignation.resignation_type,
+            "transferring_shares_to": resignation.transferring_shares_to,
+            "paid_out": resignation.paid_out,
+        }
+        form = MembershipResignationForm(data=data)
+        self.assertTrue(form.is_valid())
 
-    def test_validations(self):
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_duplicates
+    def test_validate_share_owner(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(share_owner=share_owner)
+        form = MembershipResignationForm(data={"share_owner": resignation.share_owner})
+        form.validate_share_owner(resignation.share_owner)
+        self.assertIn("share_owner", form.errors.keys())
+        self.assertIn(
+            "This member is already resigned.",
+            form.errors["share_owner"],
+        )
+
+    def test_validate_transfer_choice(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(
+            share_owner=share_owner,
+            resignation_type=MembershipResignation.ResignationType.TRANSFER,
+        )
+        form = MembershipResignationForm(
+            data={
+                "share_owner": resignation.share_owner,
+                "resignation_type": resignation.resignation_type,
+                "transferring_shares_to": resignation.transferring_shares_to,
+            }
+        )
+        form.validate_transfer_choice(
+            resignation.resignation_type,
+            transferring_shares_to=None,
         )
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_share_owner
+        self.assertIn("transferring_shares_to", form.errors.keys())
+        self.assertIn(
+            "Please select the member that the shares should be transferred to.",
+            form.errors["transferring_shares_to"],
         )
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_transfer_choice
+        form.validate_transfer_choice(
+            resignation_type=MembershipResignation.ResignationType.BUY_BACK,
+            transferring_shares_to=resignation.transferring_shares_to,
+        )
+        self.assertIn("transferring_shares_to", form.errors.keys())
+        self.assertIn(
+            "If the shares don't get transferred to another member, this field should be empty.",
+            form.errors["transferring_shares_to"],
         )
-        self.assertRaises(ValidationError, MembershipResignationForm.validate_if_gifted)
 
-    def test_membershipResignationForm_loggedInAsMemberOffice_accessGranted(self):
-        self.login_as_member_office_user()
-        resignation_form: MembershipResignationForm = MembershipResignationForm.create()
+    def test_validate_duplicates(self):
```

Comment:
Missing positive case test: no error if the resigned member and the receiving member are separate.